### ISSUE TITLE
Remove transaction from queue or batch-send mutation

### ISSUE THREAD

**Issue by Globegitter** [facebook/relay#666]
We have a use-case in our app, where we are using one (big) mutation right now to create a new user. That is done via a multi-page form and at each stage of this form we want to cache what the user entered locally (using the optimisticResponse) in case they go back etc. So for that we are using `applyUpdate` which works well.

But at the end of the multi-page form we want to send one http request which sends the whole mutation (i.e. all the data in the cache) to the server. For that we are currently using `update` to send off a new mutation with all the newly cached data. But at this point the old transactions still 'linger' in the queue.

After discussing internally, what would be good to have for us, is either some way of batching/merging theses transactions into one single request or simply being able to remove transactions from the queue without reverting their optimisticResponse.

Without knowing too much of the source-code the second method seems to me like it would be quite a bit simpler. But in any case I would be happy to PR any of the two solutions, or any other solution that would be able to full-fill this use-case.

**Comment by devknoll** [comment]
@Globegitter Is there a reason why something like the following wouldn't work:

  1. Keep track of all of the `RelayMutationTransaction`'s you generate
  2. After `onSuccess` of your actual mutation, call `rollback` on the list

This assumes that your actual mutation's fat query includes the changed fields.

FWIW in a past project of mine with a similar UI, I decided to send off incremental updates like this for UX and analytics, e.g. in the case the browser was closed without submitting.
